### PR TITLE
Allow entering DocService demo mode using a Gradle property

### PR DESCRIPTION
**Pull request by trustin**
Motivation:

When launching a `*DocServiceTest` from Gradle via IntelliJ, the
`DOC_SERVICE_DEMO` environment variable is not propagated to the test
cases.

Modifications:

- Update the build script so that the `DOC_SERVICE_DEMO` environment
  variable is set when a user passes the `-Pdemo` option to Gradle.

Result:

- More convenient testing

### CODE REVIEW

**Comment by anuraaga** [comment]
Do you mean when using Gradle runner to run the test? Hmm, it works fine for me (a relatively, but not that, think late last year, recent version of IntelliJ fixed the env variable propagation for Gradle runner).

This sort of setup works fine for me
image

**Comment by trustin** [comment]
That didn't work for me with IntelliJ IDEA EAP 2019.3, but let me try again :thinking:

**Comment by anuraaga** [comment]
FWIW I haven't tried with EAP, hope it's not broken there...

File: build.gradle [hunk 1]
```diff
@@ -47,6 +47,10 @@ allprojects {
         if (project.hasFlags('coverage')) {
             systemProperty 'com.linecorp.armeria.testing.coverage', 'true'
         }
+        // Pass special environment variable to enable demo mode for `*DocServiceTest`.
+        if (rootProject.hasProperty('demo')) {
```
Comment (ikhoon) [inline build.gradle]: nit: Mostly LGTM, but I am worried `demo` has a too wide meaning. How about renaming to  `docservice.demo`?

**Comment by trustin** [comment]
Tried again today and it seems to work fine with the env var. :sweat_smile:
